server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java

```diff
@@ -117,6 +118,12 @@ public void writeFile(StoreFileMetadata fileMetadata, long readSnapshotFileBuffe
         }
     }
 
+    private IndexOutput createIndexOutput(String tempFileName, StoreFileMetadata fileMetadata) throws IOException {
+        return Objects.equals(fileMetadata.checksum(), StoreFileMetadata.UNAVAILABLE_CHECKSUM)
```

Comment:
Instead of relying on UNAVAILABLE_CHECKSUM as a way to know if checksum must be verified or not, I would have expected to pass a flag (`useChecksums` or `verifyChecksums`) to the `MultiFileWriter` constructor indicating if checksum must be verified or not. 

And only set that flag to false where appropriate